app/src/main/java/com/learnitgirl/myvoc/database/DictionaryDBHelper.java

```diff
@@ -79,41 +81,101 @@ public boolean insert(String foreignWord, String nativeWord, String knowledge) {
+
+        String[] whereArgs = new String[]{};
+        Cursor cursor = dbRead.rawQuery(
+                "select * from " + DictionaryContract.DictionaryEntry.TABLE_NAME, whereArgs);
+
+        return cursor;
+    }
+
+    public SimpleCursorAdapter getWordsAdapter(Context context) {
+        Cursor cursor = getWords();
+        cursor.moveToFirst();
+        String[] from = {DictionaryContract.DictionaryEntry._ID, DictionaryContract.DictionaryEntry.COLUMN_NAME_FOREIGN_WORD,
+                DictionaryContract.DictionaryEntry.COLUMN_NAME_NATIVE_WORD,
+                DictionaryContract.DictionaryEntry.COLUMN_NAME_KNOWLEDGE};
+
+        int[] toViews = {R.id.wordIDTextView, R.id.foreignWordTextView, R.id.nativeWordTextView, R.id.knowledgeTextView};
+        return new SimpleCursorAdapter(context, R.layout.row, cursor, from, toViews, 0);
+    }
+
+    public boolean deleteWord(String id) {
+
+        String[] whereArgs = {id};
+        if (dbWrite.delete(DictionaryContract.DictionaryEntry.TABLE_NAME, DictionaryContract.DictionaryEntry._ID + " = ?", whereArgs) > 0) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public String getNativeWord(String foreignWord) {
+        String[] whereArgs = new String[]{DictionaryContract.DictionaryEntry.COLUMN_NAME_FOREIGN_WORD, foreignWord};
+        Cursor cursor = dbRead.rawQuery(
+                "SELECT " + DictionaryContract.DictionaryEntry.COLUMN_NAME_NATIVE_WORD +
+                        " FROM " + DictionaryContract.DictionaryEntry.TABLE_NAME +
+                        " WHERE ? = ?", whereArgs);
+
+        cursor.moveToFirst();
+        String nativeWord = "";
+        int columnIndex = cursor.getColumnIndex(DictionaryContract.DictionaryEntry.COLUMN_NAME_FOREIGN_WORD);
+        if (columnIndex != -1) {
+            nativeWord = cursor.getString(columnIndex);
+        }
+
+        return nativeWord;
+    }
+
     public String getForeignWord(String nativeWord) {
-        SQLiteDatabase db = this.getReadableDatabase();
 
-        String[] selectionArgs = {DictionaryContract.DictionaryEntry.COLUMN_NAME_NATIVE_WORD, nativeWord};
-        Cursor cursor = db.rawQuery(
+        String[] whereArgs = new String[]{DictionaryContract.DictionaryEntry.COLUMN_NAME_NATIVE_WORD, nativeWord};
+        Cursor cursor = dbRead.rawQuery(
                 "SELECT " + DictionaryContract.DictionaryEntry.COLUMN_NAME_FOREIGN_WORD +
                         " FROM " + DictionaryContract.DictionaryEntry.TABLE_NAME +
-                        " WHERE ? = ?", selectionArgs);
+                        " WHERE ? = ?", whereArgs);
 
         cursor.moveToFirst();
         String foreignWord = "";
         int columnIndex = cursor.getColumnIndex(DictionaryContract.DictionaryEntry.COLUMN_NAME_NATIVE_WORD);
-        if(columnIndex != -1) {
+        if (columnIndex != -1) {
             foreignWord = cursor.getString(columnIndex);
         }
 
         return foreignWord;
     }
```

Comment:
there is some repetition in getForeignWord and getNativeWord which could be factored out :)